bindings/pyroot_experimental/cppyy/CPyCppyy/CMakeLists.txt

```diff
@@ -46,46 +46,55 @@ set(sources
     src/Utility.cxx
 )
 
-add_library(cppyy SHARED ${headers} ${sources})
+foreach(val RANGE ${how_many_pythons})
+  list(GET python_include_dirs ${val} python_include_dir)
+  list(GET python_version_strings ${val} python_version_string)
+  list(GET python_libraries ${val} python_library)
 
-if(NOT MSVC)
-  target_compile_options(cppyy PRIVATE
-    -Wno-shadow -Wno-strict-aliasing)
-endif()
-if(NOT "${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
-  target_compile_options(cppyy PRIVATE
-    -Wno-unused-but-set-parameter)
-endif()
+  set(libname cppyy${python_version_string})
 
-# Disables warnings coming from PyCFunction casts
-if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" AND ${CMAKE_CXX_COMPILER_VERSION} VERSION_GREATER_EQUAL 8)
-  target_compile_options(cppyy PRIVATE -Wno-cast-function-type)
-endif()
+  add_library(${libname} SHARED ${headers} ${sources})
 
-# Disables warnings originating from deprecated register keyword in Python
-if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" AND CMAKE_CXX_STANDARD GREATER_EQUAL 11)
-    target_compile_options(cppyy PRIVATE -Wno-register)
-endif()
-if("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang" AND CMAKE_CXX_STANDARD GREATER_EQUAL 11)
-    target_compile_options(cppyy PRIVATE -Wno-register)
-    target_compile_options(cppyy PRIVATE -Wno-deprecated-register)
-endif()
+  if(NOT MSVC)
+    target_compile_options(${libname} PRIVATE
+      -Wno-shadow -Wno-strict-aliasing)
+  endif()
+  if(NOT "${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
+    target_compile_options(${libname} PRIVATE
+      -Wno-unused-but-set-parameter)
+  endif()
 
-# Disables warnings due to new field tp_vectorcall in Python 3.8
-if(NOT MSVC AND ${PYTHON_VERSION_STRING} VERSION_GREATER_EQUAL "3.8")
-  target_compile_options(cppyy PRIVATE -Wno-missing-field-initializers)
-endif()
+  # Disables warnings coming from PyCFunction casts
+  if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" AND ${CMAKE_CXX_COMPILER_VERSION} VERSION_GREATER_EQUAL 8)
+    target_compile_options(${libname} PRIVATE -Wno-cast-function-type)
+  endif()
 
-target_include_directories(cppyy PRIVATE ${CMAKE_BINARY_DIR}/include) # needed for string_view backport
+  # Disables warnings originating from deprecated register keyword in Python
+  if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" AND CMAKE_CXX_STANDARD GREATER_EQUAL 11)
+    target_compile_options(${libname} PRIVATE -Wno-register)
+  endif()
+  if("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang" AND CMAKE_CXX_STANDARD GREATER_EQUAL 11)
+    target_compile_options(${libname} PRIVATE -Wno-register)
+    target_compile_options(${libname} PRIVATE -Wno-deprecated-register)
+  endif()
 
-target_include_directories(cppyy PUBLIC ${PYTHON_INCLUDE_DIRS}
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/inc>
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>)
-target_link_libraries(cppyy cppyy_backend ${PYTHON_LIBRARIES})
+  # Disables warnings due to new field tp_vectorcall in Python 3.8
```

Comment:
I think there is separate PR for this: https://github.com/root-project/root/pull/5158
Make sure you don't resolve the same thing in a different places...

Reply:
@etejedor it should be fine because here I just reshuffle while in the other PR it's changed, can you confirm?

Reply:
@oshadura , @maxgalli is not introducing that change, just moving it. The disabling of those warnings for Python3.8 was introduced by @stwunsch here: 9a540f30